lib/fast_jsonapi/serialization_core.rb

```diff
@@ -65,55 +64,102 @@ def meta_hash(record, params = {})
         FastJsonapi.call_proc(meta_to_serialize, record, params)
       end
 
+      def generate_cache_key(record, namespace)
+        # using ':' as delimiters
+        [
+          namespace,
+          record.cache_key
+        ].join(':')
+      end
+
       def record_hash(record, fieldset, includes_list, params = {})
         if cache_store_instance
-          cache_opts = record_cache_options(cache_store_options, fieldset, includes_list, params)
-          record_hash = cache_store_instance.fetch(record, **cache_opts) do
-            temp_hash = id_hash(id_from_record(record, params), record_type, true)
-            temp_hash[:attributes] = attributes_hash(record, fieldset, params) if attributes_to_serialize.present?
-            temp_hash[:relationships] = relationships_hash(record, cachable_relationships_to_serialize, fieldset, includes_list, params) if cachable_relationships_to_serialize.present?
-            temp_hash[:links] = links_hash(record, params) if data_links.present?
-            temp_hash
+          cache_opts = record_cache_options(record, cache_store_options, fieldset, includes_list, params)
+          name_space = cache_opts.delete(:namespace)
+          cache_key = generate_cache_key(record, name_space)
```

Comment:
how is this different from the default ASupport implementation?

Reply:
i see. AS doesn't support multi read with different namespaces.

Reply:
maybe add a comment here explain why we don't use the built-in namespace option

Reply:
yep

Reply:
Can you please elaborate a bit on why do we need to use different namespaces?

Reply:
Okay, I see. For some mysterious reason `jsonapi-serializers` uses CacheStore's parameter `namespace` to distinguish cache keys for a same endpoint and data queried with different fieldsets, instead of making it a part of the key. That's quite surprising and certainly unexpected for sure, but still I don't think it's possible to get several different values for namespaces in the particular serialiser processing a single requests.

Reply:
good call, that case it's also quite interesting. The main reason is the params options passed in serializer could result different values. For example, you can check endpoints `Api::V2::MealInstanceController#fetch` and `Api::OnDemand::V2::MealInstanceController#fetch`.